### DISCUSSION TITLE
posts/2023/09/sound-reasoning/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#34] [Announcements]
# posts/2023/09/sound-reasoning/

Although the fabled Year of the Linux Desktop probably will never happen, a simple (?) kernel tweak could do wonders for desktop adoption.

https://www.brycewray.com/posts/2023/09/sound-reasoning/

**Comment by davidsneighbour** [comment]
How to separate "true" developers from wannabe developers:

Wannabe Developer: “Well, it works on my machine.”
True Developer: "I wonder why it works on my machine."

Reading your "Apple on Linux" posts I am always reminded of those people that migrate "Doom" to fridges and "smart" toothbrushes. It works. Is is supposed to work? Maybe not. **Maybe ask those that develop the hardware for working drivers for that hardware instead of asking those that manage to keep them together.** Those soundchips not developed by Apple probably adhere to a certain system to process their sound. 

Try getting those Windows-RGB-Blink-RAM-modules to blink under Ubuntu :) Someone should do something about that :)

**Reply by brycewray** [reply]
An Intel Mac is still just a computer with an Intel CPU, not some oddball IoT device. I concede (and noted as much in a [previous post](https://www.brycewray.com/posts/2023/09/new-life-old-mac-linux-follow-up/)) that Apple made some questionable choices regarding how an Intel Mac handles audio — and video, for that matter — **but**, still, the growing quantity of “high-and-dry” Intel Macs constitutes an unusual opportunity to grow Linux on the desktop ***if*** there’s a kernel-level fix for this one particular glitch.

**Reply by davidsneighbour** [reply]
It's not a glitch. It's a business strategy. I recently saw an article, can't find it now, but will link it after I found it, about how refrigerators worked for many years when they were built in the 60s/70s. After that, sales departments realized, that offering a product that is usable for many years doesn't make much money, except for the first time. Apple is a company that embraces this strategy. Their products are maybe hip for 2 years (if at all), properly supported for 6 to 8 years, and then if you don't buy a new fancy product of theirs, you are a failed customer. Fridges too. 

That article explained it much better. But a Unix/Linux compatible computer (or let's say hardware that is composed together and runs on open source operating) systems will run on those systems in 10 years... in 20 years... etc. The only thing you have to live with is when new features are developed or discovered. They might not work on old hardware. 

And by compatible, I don't mean "Hey, it's Intel" but "We, as producers of this computer certify compatibility with this and that OS by having it tested with the current drivers". 

My main point was that you kneel down to the wrong people. If you want audio on Apple products to work, you need to go to Apple people. The funny (for me as a non-apple person) thing is, that it's probably "just" a `1` or `true` somewhere. 

As in:

- the taxi driver does not know how to fix that knocking in your Kia
- the sheepdog does not produce wool
- the SSG you are using is not writing your content for you

;)
